real_estate_module/models/estate_property.py

```diff
@@ -0,0 +1,191 @@
+        ],
+        help="The orientation of the garden"
+    )
+    
+    active = fields.Boolean(
+        string="Active",
+        default=True,
+        help="Indicates whether the property is active and available for sale"
+    )
+    
+    state = fields.Selection(
+        string="Current State",
+        selection=[
+            ('new', 'New'),
+            ('offer_received', 'Offer Received'),
+            ('offer_accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled')
+        ],
+        required=True,
+        copy=False,
+        default='new',
+        help="The current state of the property"
+    )
+    
+    property_type_id = fields.Many2one(
+        comodel_name='estate.property.type',
+        string="Property Type",
+        help="The type of the property (e.g., apartment, house, etc.)"
+    )
+    
+    buyer = fields.Many2one(
+        comodel_name='res.partner',
+        string="Buyer",
+        help="The buyer who purchased the property"
+    )
+    
+    salesman = fields.Many2one(
+        comodel_name='res.users',
+        string="Salesman",
+        default=lambda self: self.env.user,
+        help="The salesperson responsible for the property"
+    )
+    
+    tag_ids = fields.Many2many(
+        comodel_name='estate.property.tags',
+        string="Tags",
+        required=True,
+        help="Tags associated with the property (e.g., luxury, beachfront)"
+    )
+    
+    offer_ids = fields.One2many(
+        comodel_name='estate.property.offer',
+        inverse_name="property_id",
+        help="Offers made for this property"
+    )
+    
+    total_area = fields.Float(
+        compute='_compute_total_area',
+        help="The total area of the property (living area + garden area)"
+    )
+    
+    best_price = fields.Float(
+        compute='_compute_best_price',
+        help="The best price offered for the property"
+    )
+    
+    status = fields.Selection(
+        string="Status",
+        selection=[
+            ('cancel',"Cancelled"),
+            ('sold',"Sold")
+        ]
+    )
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for property in self:
+            property.total_area = property.living_area + property.garden_area
+    
+    @api.depends('offer_ids.price')
+    def _compute_best_price(self):
+        for property in self:
+            if property.offer_ids:
+                property.best_price = max(property.offer_ids.mapped('price'))
+            else:
+                property.best_price = 0.0
+
+    @api.onchange('garden')
+    def _onchange_garden_area_orientation(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = ''
+    
+    def action_cancel_button(self):
+        if not self.status:
+            self.status='cancel'
```

Comment:
```suggestion
            self.status = 'cancel'
```
make sure to have empty space before using any operator before and after it